### DISCUSSION TITLE
Use of pds.description, pds.name, and pds.comment in Discipline Namespaces

### DISCUSSION THREAD

**Discussion by jordanpadams** [pds-data-dictionaries/PDS-Data-Dictionaries.github.io#16] [Ideas]
Per offline discussions with @acraugh, we have come up with another plan for address use of common attributes in Discipline Namespaces for inheriting data types.

**_Near-term:_** We actually did not come to a conclusion here, but I propose we continue on as status quo until Final/Ideal solution is ready.

**_Final/Ideal:_** New SCR to create new PDS_DESCRIPTION, PDS_NAME, and PDS_COMMENT data types, instead of the current paradigm of inheriting the attribute data type. This is how we should be re-using attribute data types in PDS4.

**Comment by mdrum** [comment]
Interesting. It sets a new non-backwards-compatible precedent for ways to reuse common attributes across the PDS. 

I'd prefer to use the attributes themselves, but I can live with this. We're trading one technical hurdle for another. 

Would this mean that if there were other attributes that we would like to have standardized and reused, we would also need to convert those into data types? For example, the msn namespace currently copies the start_local_true_solar_time from the global namespace; I was hoping to deduplicate that.

**Comment by acraugh** [comment]
The whole point of the extended discussion was that the only thing the pds
attributes represent is a re-naming of an existing data type in the first
place. We have a defined method for renaming existing data types for use in
designing namespaces. We're eliminating the undocumented, back-door method
in favor of the existing, documented method.

-Anne.

**Comment by mdrum** [comment]
I would like to save us the future data usability hassle by preserving common semantic information though. Any software service that integrates with PDS data will want to know that certain attributes represent specific meaningful value types. A name is semantically different from a description even if they have the same data type, and any interface would want to be able to present those differently.

I would like there to be a way to recognize and implement more common value types in the future.

**Comment by acraugh** [comment]
If you want data usability based on semantic definitions, then the definitions need to be made in a semantically actionable way - that is,  correlating the programmatic definition to the human definition so that the semantic information can be used to effectively present information to humans as well as to guide programmatic processing outside of human intervention. 

Here's what we actually have in the information model for these three things:

**pds:name** - a UTF-8 string with normalized whitespace and a length restriction of 0 < _length_ < 256 characters. (This is identical to the definitions of five other attributes, including "keyword", "title", and "unit". So there is no semantic specificity inherent in the data type definition associated with "name".) The human definition is: 

> The name attribute provides a word or combination of words by which the Agency is  known.

which pretty clearly indicates it is only supposed to be used in the context of the &lt;Agency&gt; class. Needless to say, this apparent restriction is not enforced.

**pds:comment** - an ASCII string with no whitespace processing and a minimum length requirement of 1 character. (This is identical to the definitions of 22 other attributes, including "abstract_desc", "citation_text", and "mission_desc".) The human definition is:

> The comment attribute is a character string expressing one or more remarks or thoughts relevant to the object.

which implies it should only ever be used within the context of an object-description class. That is very nebulously defined within PDS documents, but given the usage in the Information Model document it would seem to indicate that it should not be used in any class that does not directly reference an object (digital or non-digital), and that the content should be directly relevant to that object. Depending on how "object" is (re-)interpreted, that seems to preclude many of the uses to which it is currently applied, both in the core and in discipline namespaces.

**pds:description** - a UTF-8 string with no whitespace processing and a minimum length requirement of 1 character. (This is identical to the definitions of 5 other attributes, including "author_list", "definition", and "document_name".) The human definition is:

> The description attribute provides a statement, picture in words, or account that describes or is otherwise relevant to the object.

which, once again, implies it should only ever be used in an object-description context and should only ever address the object itself. In the core, "description" does routinely show up in an object-description context, although it also shows up in other contexts. In discipline namespaces, usage is more...free.

So what I am and have been arguing is that these attributes are neither defined nor used in a semantically significant or consistent way with respect to either natural language or programming constructs. Treating them as though they have any semantic significance is deceptive. If you attempt to apply semantic significance at this point, then you are facilitating misinterpretation of products archived prior to that imposition.

What's the alternative? There are two - either define a data type which can carry some semantic clues and start to isolate these concepts from the other concepts with identical programmatic definitions, or redefine the above to have semantically significant definitions and deal with the consequences. Specifically: 

pds:name should go away. It is used to mean too many different things, from titles of agencies to names of facilities to identifiers of Fields in Tables. Names are significant and should be defined and treated contextually. At the least, names of humans should be distinguished from names of non-humans so that some format requirements can be applied. A "name" data type might be used as the basis for definition, but in attributes "name" should probably always come with a qualifier and the appropriate validation.

pds:comment should be clearly distinguished from pds:description in the human definition. Typically, a "description" is essential information, while a "comment" is not. It is not reasonable, semantically, to force an author to identify text as a "description" rather than a "comment" merely because they need access to non-ASCII characters. Both descriptions and comments typically apply to their immediate surroundings, not something referenced elsewhere in the label. For programmatic support, there are expectations that should be set and, potentially, validation that should be applied. It should be noted, for example, that writers of descriptions and comments should not rely on tab settings or font selection to ensure readability of their text. They should also be informed that certain characters must be escaped, and that they may not include HTML or any other sort of markup in their text - all of which should be spot-checked in validation. If we do want to accommodate things like ASCII art or columnar tables or markup imbedded in otherwise free-form text then some serious design work is needed - and programs will need some clear indication when this should be expected as a possibility.

**Comment by mdrum** [comment]
Definitely all fair points. The attributes as built aren't appropriate for this. Clearly there's a desire to have something _like_ them in place though.

You've also pointed out fairly that attributes should have appropriate contextual definitions. So, actually creating bespoke data types for these sorts of things might actually be more appropriate, as long as those data types are surfaced to data/API consumers more readily.

@jordanpadams do you think this is a possible with the new registry?
